app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java

```diff
@@ -46,4 +49,17 @@ private void validatePassword(String password) {
             validCredentials = false;
         }
     }
+
+    public void onSignupButtonClick() {
+        getView().navigateToSignUpPage();
+    }
+
+    public void onTermsConditionsLinkClick(String url) {
+        if (!url.startsWith(protocol)) {
+            url = protocol + url;
+        }
+        Uri uri = Uri.parse(url);
+        Intent intent = new Intent(Intent.ACTION_VIEW, uri);
+        getView().openBrowser(intent);
+    }
```

Comment:
this should be in the fragment. Remember that the presenter shouldn't know about the "context" of the app